scm/github.go

```diff
@@ -717,26 +601,130 @@ func (s *GithubSCM) CreateCourse(ctx context.Context, opt *NewCourseOptions) ([]
-	// Create repo, or return existing repo if it already exists;
-	// if repo is found, it is safe to reuse it.
+// UpdateEnrollment updates organization and team membership and creates user repositories.
+func (s *GithubSCM) UpdateEnrollment(ctx context.Context, opt *UpdateEnrollmentOptions) (*Repository, error) {
+	if !opt.valid() {
+		return nil, ErrMissingFields{
+			Method:  "UpdateEnrollment",
+			Message: fmt.Sprintf("%+v", opt),
+		}
+	}
+	org, err := s.GetOrganization(ctx, &GetOrgOptions{
+		Name: opt.Organization,
+	})
+	if err != nil {
+		return nil, err
+	}
+	switch opt.Status {
+	case qf.Enrollment_STUDENT:
+		// give access to the course's info and assignments repositories
+		if err := s.grantAccessToCourseRepos(ctx, org.Name, opt.User); err != nil {
+			return nil, err
+		}
+		// add student to the organization's "students" team
+		if err := s.addUserToStudentsTeam(ctx, org.Name, opt.User); err != nil {
+			return nil, err
+		}
+		return s.createStudentRepo(ctx, org, opt.User)
+	case qf.Enrollment_TEACHER:
+		// promote to organization owners
+		role := OrgOwner
+		if _, _, err := s.client.Organizations.EditOrgMembership(ctx, opt.User, org.Name, &github.Membership{Role: &role}); err != nil {
+			return nil, err
+		}
+		err = s.promoteToTeachers(ctx, org.Name, opt.User)
+	}
+	return nil, err
+}
+
+// RejectEnrollment removes user's repository and revokes user's membersip in the course organization.
+func (s *GithubSCM) RejectEnrollment(ctx context.Context, opt *RejectEnrollmentOptions) error {
+	if !opt.valid() {
+		return ErrMissingFields{
+			Method:  "RejectEnrollment",
+			Message: fmt.Sprintf("%+v", opt),
+		}
+	}
+	org, err := s.GetOrganization(ctx, &GetOrgOptions{
+		ID: opt.OrganizationID,
+	})
+	if err != nil {
+		return err
+	}
+	if _, err := s.client.Organizations.RemoveMember(ctx, org.Name, opt.User); err != nil {
+		return err
+	}
+	return s.DeleteRepository(ctx, &RepositoryOptions{ID: opt.RepositoryID})
+}
+
+// RevokeTeacherStatus removes user from teachers team, revokes owner status in the organization.
+func (s *GithubSCM) RevokeTeacherStatus(ctx context.Context, opt *UpdateEnrollmentOptions) error {
+	if _, err := s.client.Teams.RemoveTeamMembershipBySlug(ctx, opt.Organization, TeachersTeam, opt.User); err != nil {
+		return err
+	}
+	if _, _, err := s.client.Teams.AddTeamMembershipBySlug(ctx, opt.Organization, StudentsTeam, opt.User,
+		&github.TeamAddTeamMembershipOptions{Role: TeamMember}); err != nil {
+		return err
+	}
+	role := OrgMember
+	_, _, err := s.client.Organizations.EditOrgMembership(ctx, opt.User, opt.Organization, &github.Membership{Role: &role})
+	return err
+}
+
+// createStudentRepo creates {username}-labs repository and provides pull/push access to it for the given student.
+func (s *GithubSCM) createStudentRepo(ctx context.Context, org *qf.Organization, login string) (*Repository, error) {
```

Comment:
We only need `org.Name` in this func; suggest to replace `qf.Organization` with `organizationName, login string` or something like that.